firedrake/assemble.py

```diff
@@ -1628,7 +1628,7 @@ def _integral_type(self):
 
     @cached_property
     def _mesh(self):
-        return self._form.ufl_domains()[self._kinfo.domain_number]
```

Comment:
@ksagiyam `ufl_domains()` returns a `set`, so I casted it into a tuple, how is this going to affect `MixedMesh`?

Reply:
That seems wrong. Can the set be sorted in some way?

Reply:
I think this has been fixed recently:
https://github.com/FEniCS/ufl/blob/d8350119ac162caf949f8aad6cee76223a5c637f/ufl/domain.py#L201

I will need to rebase my branch, but you do not need to worry about that.

Reply:
I encountered the issue with SLATE tensors (fix in this PR), and maybe FormSum ([fix in UFL](https://github.com/FEniCS/ufl/pull/325)). join_domains returns a set, so we need to call sort_domains afterwards,